crates/revm/src/optimism/executor.rs

```diff
@@ -0,0 +1,313 @@
+//! Optimism-specific execution logic.
+
+use crate::{
+    executor::{verify_receipt, Executor},
+    to_reth_acc,
+};
+use reth_interfaces::executor::{BlockExecutionError, BlockValidationError};
+use reth_primitives::{bytes::BytesMut, Address, Block, Hardfork, Receipt, U256};
+use reth_provider::{BlockExecutor, PostState, StateProvider};
+use reth_revm_primitives::into_reth_log;
+use revm::primitives::ResultAndState;
+use std::sync::Arc;
+
+impl<DB> BlockExecutor<DB> for Executor<DB>
+where
+    DB: StateProvider,
+{
+    fn execute(
+        &mut self,
+        block: &Block,
+        total_difficulty: U256,
+        senders: Option<Vec<Address>>,
+    ) -> Result<PostState, BlockExecutionError> {
+        let (post_state, cumulative_gas_used) =
+            self.execute_transactions(block, total_difficulty, senders)?;
+
+        // Check if gas used matches the value set in header.
+        if block.gas_used != cumulative_gas_used {
+            return Err(BlockValidationError::BlockGasUsed {
+                got: cumulative_gas_used,
+                expected: block.gas_used,
+            }
+            .into())
+        }
+
+        self.apply_post_block_changes(block, total_difficulty, post_state)
+    }
+
+    fn execute_and_verify_receipt(
+        &mut self,
+        block: &Block,
+        total_difficulty: U256,
+        senders: Option<Vec<Address>>,
+    ) -> Result<PostState, BlockExecutionError> {
+        let post_state = self.execute(block, total_difficulty, senders)?;
+
+        // TODO Before Byzantium, receipts contained state root that would mean that expensive
```

Comment:
Can we remove the `TODO` here?
```suggestion
        // Before Byzantium, receipts contained state root that would mean that expensive
```